tensorflow/compiler/mlir/xla/transforms/map_lhlo_to_scalar_op.h

```diff
@@ -148,44 +155,173 @@ inline Optional<CmpIPredicate> getIntCmpPredicate(
 }
 
 template <>
-inline Operation* MapLhloOpToStdScalarOp<xla_lhlo::CompareOp>(
+inline Value* MapLhloOpToStdScalarOp<xla_lhlo::CompareOp>(
     xla_lhlo::CompareOp lhlo_op, ArrayRef<Type> result_types,
     ArrayRef<Value*> block_args, OpBuilder b) {
   const auto& lhs = block_args[0];
   const auto& rhs = block_args[1];
   Type element_type = lhs->getType();
+  Operation* op;
   if (element_type.isa<IntegerType>()) {
     Optional<CmpIPredicate> predicate =
         getIntCmpPredicate(lhlo_op.comparison_direction());
-    assert(predicate.hasValue() && "expected valid comparison direction");
-    return b.create<ScalarIOp<CompareOp>>(lhlo_op.getLoc(),
-                                          predicate.getValue(), lhs, rhs);
-  }
-  if (element_type.isa<FloatType>()) {
-    return b.create<ScalarFOp<CompareOp>>(
+    assert(predicate.hasValue() && "expected valid comparision direction");
+    op = b.create<ScalarIOp<CompareOp>>(lhlo_op.getLoc(), predicate.getValue(),
+                                        lhs, rhs);
+  } else if (element_type.isa<FloatType>()) {
+    op = b.create<ScalarFOp<CompareOp>>(
         lhlo_op.getLoc(), getFloatCmpPredicate(lhlo_op.comparison_direction()),
         lhs, rhs);
   }
-  return nullptr;
+  return (op) ? op->getResult(0) : nullptr;
 }
 
 template <>
-inline Operation* MapLhloOpToStdScalarOp<xla_lhlo::SelectOp>(
+inline Value* MapLhloOpToStdScalarOp<xla_lhlo::SelectOp>(
     xla_lhlo::SelectOp lhlo_op, ArrayRef<Type> result_types,
     ArrayRef<Value*> block_args, OpBuilder b) {
-  return b.create<::mlir::SelectOp>(lhlo_op.getLoc(), result_types, block_args,
-                                    mlir::None);
+  Operation* op = b.create<::mlir::SelectOp>(lhlo_op.getLoc(), result_types,
+                                             block_args, mlir::None);
+  return op->getResult(0);
+}
+
+template <>
+inline Value* MapLhloOpToStdScalarOp<xla_lhlo::AbsOp>(
+    xla_lhlo::AbsOp lhlo_op, ArrayRef<Type> result_types,
+    ArrayRef<Value*> block_args, OpBuilder b) {
+  Type element_type = block_args.front()->getType();
+  if (element_type.isa<FloatType>()) {
+    Operation* op = b.create<::mlir::AbsFOp>(lhlo_op.getLoc(), result_types,
+                                             block_args, mlir::None);
+    return op->getResult(0);
+  }
+  return nullptr;
 }
 
 template <>
-inline Operation* MapLhloOpToStdScalarOp<xla_lhlo::ExpOp>(
+inline Value* MapLhloOpToStdScalarOp<xla_lhlo::CeilOp>(
+    xla_lhlo::CeilOp lhlo_op, ArrayRef<Type> result_types,
+    ArrayRef<Value*> block_args, OpBuilder b) {
+  Type element_type = block_args.front()->getType();
+  if (element_type.isa<FloatType>()) {
+    Operation* op = b.create<::mlir::CeilFOp>(lhlo_op.getLoc(), result_types,
```

Comment:
Generally, avoid converting to `Operation *`. Either use auto here or the specific type (`CeilFOp`) to get access to operation specific accessors like getResult(). This comment applies to all templates here.